lib/components/DatePicker.vue

```diff
@@ -0,0 +1,43 @@
+<template>
+    <input 
+        :max="maxDate" 
+        :min="minDate" 
+        @change="dateChange($event)" 
+        :value="calcDate" 
+        type="date"
+    />
+</template>
+
+<script>
+export default {
+    name: 'DatePicker',
+    props: [
+        'model',
+        'day',
+        'month',
+        'year',
+        'minDate',
+        'maxDate',
+        'dateChange'
+    ],
+    computed: {
+        calcDate: function () {
+            if(this.year != undefined && this.month != undefined && this.day != undefined) {
```

Comment:
You can remove "!= undefined" as none of these values can have 0 value. Also, always use === or !===